### ISSUE TITLE
Go-to definition jumps to hs-boot files

### ISSUE THREAD

**Issue by mpickering** [digital-asset/ghcide#509]
If an identifier is defined in a hs-boot file then jumping to the definition will jump into the hs-boot file.

Technically, this is where the definition is, but really it would be better to jump to the definition in 
the source file I think.

**Comment by ndmitchell** [comment]
Agreed - jumping to hs-boot files doesn't seem like the right thing to do in most cases.

**Comment by jneira** [comment]
I've not observed this nowadays, is it still an issue?

**Comment by sgraf812** [comment]
I'm observing this a lot in GHC's code base. Any reference to `GHC.Types.Id.idSpecialisation`, for example. `GHC.Types.Id` has an hs-boot file.

**Comment by wz1000** [comment]
#898 will fix this
